datadog/resource_datadog_synthetics_test_.go

```diff
@@ -1127,6 +1187,101 @@ func buildSyntheticsAPITestStruct(d *schema.ResourceData) *datadogV1.SyntheticsA
 
 	config.SetConfigVariables(configVariables)
 
+	if attr, ok := d.GetOk("api_step"); ok && syntheticsTest.GetSubtype() == "multi" {
```

Comment:
all of this seems very similar to what's done above in the case of a single step test. Can we factorize it ?

Reply:
Given that there are different models (`SyntheticsAPITest` / `SyntheticsAPIStep`) I'm not sure we can factorize everything? But I'll see if there are more fields that we can extract in functions.